File: leetcode/509. Fibonacci Number/Solution.py
class Solution:
    def fib(self, n: int) -> int:
        if n==0: 
            return 0 
        fib_dp = list()
        fib_dp.append(0)
        fib_dp.append(1)
        for i in range(2,n+1):
            fib_dp.append(fib_dp[i-1]+fib_dp[i-2])
        return(fib_dp[n])

File: leetcode/509. Fibonacci Number/Solution.java
class Solution {
    public int fib(int n) {
        int [] fin_dp = new int[n+1];
        if(n==0) return 0;
        fin_dp[0] = 0;
        fin_dp[1] = 1;
        for(int i = 2; i <= n ; i++){
            fin_dp[i]=fin_dp[i-1]+fin_dp[i-2];
            
        }
        return(fin_dp[n]);
    }
}
